File: src/mobile/android/app/src/main/kotlin/com/example/saasportal/domain/model/ServiceUsageDetail.kt
package com.example.saasportal.domain.model

data class ServiceUsageDetail(
    val serviceCode: String,
    val serviceName: String,
    val planName: String,
    val freeTierLimit: Int,
    val overageUnitPrice: Double,
    val usageDetails: List<MonthlyUsage>,
) {
    companion object {
        fun preview(): ServiceUsageDetail = ServiceUsageDetail(
            serviceCode = "CONNECT_CHAT",
            serviceName = "ConnectChat",
            planName = "ビジネスプラン",
            freeTierLimit = 10000,
            overageUnitPrice = 1.5,
            usageDetails = MonthlyUsage.previewList(),
        )
    }
}

data class MonthlyUsage(
    val yearMonth: String,
    val metricName: String,
    val quantity: Int,
    val usageRate: Double,
    val billedAmount: Double,
    val primaryUseCase: String?,
) {
    companion object {
        fun previewList(): List<MonthlyUsage> = listOf(
            MonthlyUsage(
                yearMonth = "2025-06",
                metricName = "messages",
                quantity = 8500,
                usageRate = 85.0,
                billedAmount = 500.0,
                primaryUseCase = "社内コミュニケーション",
            ),
            MonthlyUsage(
                yearMonth = "2025-05",
                metricName = "messages",
                quantity = 7200,
                usageRate = 72.0,
                billedAmount = 500.0,
                primaryUseCase = "社内コミュニケーション",
            ),
        )
    }
}

File: src/mobile/ios/SaaSPortal/Models/ServiceUsageDetail.swift
import Foundation

/// 月次利用詳細の個別アイテム。
struct ServiceUsageDetailItem: Codable, Identifiable {
    var id: String { yearMonth }

    let yearMonth: String
    let metricName: String
    let quantity: Int
    let usageRate: Double
    let billedAmount: Double
    let primaryUseCase: String?
}

/// サービス別利用詳細レスポンス。
struct ServiceUsageResponse: Codable {
    let serviceCode: String
    let serviceName: String
    let planName: String
    let freeTierLimit: Int
    let overageUnitPrice: Double
    let usageDetails: [ServiceUsageDetailItem]
}

// MARK: - Preview Fixtures

extension ServiceUsageDetailItem {
    static let previewJan = ServiceUsageDetailItem(
        yearMonth: "2025-01",
        metricName: "messages",
        quantity: 7500,
        usageRate: 75.0,
        billedAmount: 15000,
        primaryUseCase: "社内コミュニケーション"
    )

    static let previewFeb = ServiceUsageDetailItem(
        yearMonth: "2025-02",
        metricName: "messages",
        quantity: 8200,
        usageRate: 82.0,
        billedAmount: 15000,
        primaryUseCase: "社内コミュニケーション"
    )

    static let previewMar = ServiceUsageDetailItem(
        yearMonth: "2025-03",
        metricName: "messages",
        quantity: 9100,
        usageRate: 91.0,
        billedAmount: 18200,
        primaryUseCase: nil
    )
}

extension ServiceUsageResponse {
    static let preview = ServiceUsageResponse(
        serviceCode: "CONNECT_CHAT",
        serviceName: "Connect Chat",
        planName: "Business",
        freeTierLimit: 10000,
        overageUnitPrice: 2.0,
        usageDetails: [
            .previewMar,
            .previewFeb,
            .previewJan,
        ]
    )
}
